stdlib/src/collections/list.mojo

```diff
@@ -124,6 +122,18 @@ struct List[T: CollectionElement](CollectionElement, Sized):
         for value in values:
             self.append(value[])
 
+    fn __init__(inout self: Self, owned ptr: AnyPointer[T], length: Int, capacity: Int):
+        """Constructs a list from a pointer and its length.
+
+        Args:
+            ptr: The pointer to the data.
+            length: The number of elements in the list.
+            capacity: The capacity of the list.
+        """
+        self.data = ptr
+        self.size = length
+        self.capacity = capacity
```

Comment:
capacity has to be larger then length and we probably should not expose capacity to the user in this situation. Also the capacity should be the actual size of the pointer/array and it should probably need to be `_realloc()` if necessary.

Reply:
Got it. So we should not set capacity on self right? Also then will we just derive the capacity from the pointer and then remove the capacity parameter?

Reply:
yes i believe so

Reply:
I am trying to do the same but the AnyPointer doesn't have any param as size or capacity. How do we calculate the capacity? Am a bit new to doing low level operations. Thanks

Reply:
The pointer doesn't have a capacity parameter. The user needs to know how much space they allocated—that's one reason pointers are "unsafe". 

If I understand the intention of this feature request, the method is intended to avoid a copy (for performance reasons) at the cost of some safety. So the user needs to know exactly what they're doing.

Reply:
In this case you can just take the pointer with a length. Then assign a capacity that is bigger then the length. The rest of method should be able to take care of the adjustment of sizing.

Reply:
Yes even in [from_raw_parts](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts) it is mentioned the same that the user needs to take care of all the invariant checking as it is highly unsafe.

So I think keeping the capacity taken from the user makes sense. You can check the Rust reference above. Let me know if I need to change this

Reply:
> If I understand the intention of this feature request, the method is intended to avoid a copy (for performance reasons) at the cost of some safety. So the user needs to know exactly what they're doing.

Well wouldn't it be better to just have 1 unsafe argument then 2?